spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/RabbitConnectionFactoryBean.java

```diff
@@ -481,34 +539,46 @@ protected void setUpSSL() throws Exception {
 			}
 			PathMatchingResourcePatternResolver resolver = new PathMatchingResourcePatternResolver();
 			String keyStoreName = getKeyStore();
-			Assert.state(StringUtils.hasText(keyStoreName), KEY_STORE + " property required");
 			String trustStoreName = getTrustStore();
-			Assert.state(StringUtils.hasText(trustStoreName), TRUST_STORE + " property required");
 			String keyStorePassword = getKeyStorePassphrase();
-			Assert.state(StringUtils.hasText(keyStorePassword), KEY_STORE_PASS_PHRASE + " property required");
 			String trustStorePassword = getTrustStorePassphrase();
-			Assert.state(StringUtils.hasText(trustStorePassword), TRUST_STORE_PASS_PHRASE + " property required");
 			String keyStoreType = getKeyStoreType();
 			String trustStoreType = getTrustStoreType();
-			Resource keyStore = resolver.getResource(keyStoreName);
-			Resource trustStore = resolver.getResource(trustStoreName);
-			char[] keyPassphrase = keyStorePassword.toCharArray();
-			char[] trustPassphrase = trustStorePassword.toCharArray();
-
-			KeyStore ks = KeyStore.getInstance(keyStoreType);
-			ks.load(keyStore.getInputStream(), keyPassphrase);
-
-			KeyManagerFactory kmf = KeyManagerFactory.getInstance("SunX509");
-			kmf.init(ks, keyPassphrase);
-
-			KeyStore tks = KeyStore.getInstance(trustStoreType);
-			tks.load(trustStore.getInputStream(), trustPassphrase);
-
-			TrustManagerFactory tmf = TrustManagerFactory.getInstance("SunX509");
-			tmf.init(tks);
+			char[] keyPassphrase = null;
+			if (StringUtils.hasText(keyStorePassword)) {
+				keyPassphrase = keyStorePassword.toCharArray();
+			}
+			char[] trustPassphrase = null;
+			if (StringUtils.hasText(trustStorePassword)) {
+				trustPassphrase = trustStorePassword.toCharArray();
+			}
+			KeyManager[] keyManagers = null;
+			TrustManager[] trustManagers = null;
+			if (StringUtils.hasText(keyStoreName) || this.keyStoreResource != null) {
+				Resource keyStoreResource = this.keyStoreResource != null ? this.keyStoreResource
+						: resolver.getResource(keyStoreName);
+				KeyStore ks = KeyStore.getInstance(keyStoreType);
+				ks.load(keyStoreResource.getInputStream(), keyPassphrase);
+				KeyManagerFactory kmf = KeyManagerFactory.getInstance("SunX509");
+				kmf.init(ks, keyPassphrase);
+				keyManagers = kmf.getKeyManagers();
+			}
+			if (StringUtils.hasText(trustStoreName) || this.trustStoreResource != null) {
+				Resource trustStoreResource = this.trustStoreResource != null ? this.trustStoreResource
+						: resolver.getResource(trustStoreName);
+				KeyStore tks = KeyStore.getInstance(trustStoreType);
+				tks.load(trustStoreResource.getInputStream(), trustPassphrase);
+				TrustManagerFactory tmf = TrustManagerFactory.getInstance("SunX509");
+				tmf.init(tks);
+				trustManagers = tmf.getTrustManagers();
+			}
 
+			if (this.logger.isDebugEnabled()) {
+				this.logger.debug("Initializing SSLContext with KM: " + keyManagers + ", TM: " + trustManagers
```

Comment:
Yeah - I only added this for the test (because init is final so I can't spy it). At least we can see which one is null.